app-modules/meeting-center/src/GoogleCalendarManager.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+namespace Assist\MeetingCenter;
+
+use DateTime;
+use Google\Client;
+use DateTimeInterface;
+use Google\Service\Oauth2;
+use Illuminate\Support\Carbon;
+use Google\Service\Calendar\Event;
+use Assist\MeetingCenter\Models\Calendar;
+use Google\Service\Calendar\EventDateTime;
+use Assist\MeetingCenter\Models\CalendarEvent;
+use Google\Service\Calendar as GoogleCalendar;
+use Assist\MeetingCenter\Contracts\CalendarInterface;
+
+class GoogleCalendarManager implements CalendarInterface
+{
+    /**
+     * @see https://developers.google.com/calendar/api/v3/reference/events/watch
+     *
+     * @todo watch?
+     * @todo multiple calendars?
+     * @todo multiple users? one event? multiple events?
+     * */
+    public static function type(): string
+    {
+        return 'google';
+    }
+
+    public function getEvents(Calendar $calendar, ?Datetime $start = null, ?Datetime $end = null): array
+    {
+        /**
+         * @todo create without sync?
+         * @todo sync uncreated events?
+         * */
+        $service = (new GoogleCalendar($this->client($calendar)));
+
+        $parameters = [
+            'singleEvents' => true,
+            'orderBy' => 'startTime',
+            'maxResults' => 2500,
+            'pageToken' => null,
+        ];
+
+        if (is_null($start)) {
+            $start = now()->startOfDay();
+        }
+        $parameters['timeMin'] = $start->format(DateTimeInterface::RFC3339);
+
+        // if (is_null($end)) {
+        //     $end = now()->addYear()->endOfDay();
+        // }
+        // $parameters['timeMax'] = $end->format(DateTimeInterface::RFC3339);
```

Comment:
I copied it from the Spatie package because it appears they wanted to cap how far out they got events. Ultimately it didn't seem that useful so I commented it out. I guess it depends whether you think having an implicit cap is useful or not?

Reply:
I remember us talking about this but forgot the details. Without this, how far will Google go out?

Reply:
I think forever. It uses the maxResults as the hard limit per page, but if you had 100,000 events over the next 20 years it would just get all of them.